etc/openqa/openqa.ini

```
@@ -139,6 +139,9 @@ blocklist = job_grab job_done
 
 [misc_limits]
 #untracked_assets_storage_duration = 14
+# Skips the cleanup of results/assets if the used disk space on the given partition is below the specified percentage
+#result_cleanup_min_percentage = 0
+#asset_cleanup_min_percentage = 0
```

As we commonly use the plural form I suggest for consistency to use that here as well

```suggestion
#results_cleanup_min_percentage = 0
#assets_cleanup_min_percentage = 0
```

e.g. also see the lines above and below

Also I suggested to use the "free" capacity, not "used" capacity. Any specific reason you chose "used"?

Reasons for using "free" is that commonly one wants to keep a bit of free space regardless of how much is actually used. And when one increases the available storage space the number for "used" might need adjustment but not necessarily for "free". Commonly it's a good practice to keep something like 10% of a filesystem free to provide enough buffer as well as keep a filesystem performant. This number would always stay the same, regardless of how much more one adds as storage. Of course everyone could calculate that again to be "90% used" but why ask the user to calculate that and you then calculate "100 - …" in code as well.

---

I don't care about this. I guess I've chosen it because `0` then naturally becomes the "disabled" value which is quite common. With your suggestion I'd change the default to `100` which will then be the "disabled" value.

How would you write the explaining comment then?

I suppose

> Skips the cleanup of results/assets if the used disk space on the given partition is below the specified percentage

would turn into

> Performs the cleanup of results/assets only if the free disk space on the relevant partition is below the specified percentage (and aborts early otherwise)

The keys would be e.g. `results_cleanup_max_free_percentage` then (unless you have a better idea).

---

> With your suggestion I'd change the default to 100 which will then be the "disabled" value.

Do you mean that "disabled" would "disable the feature to skip cleanup"?

> The keys would be e.g. "results_cleanup_max_free_percentage"

I thought of "results_min_free_percentage" as in "trying to keep at least the specified percentage free". And you don't need the word "cleanup" because "min_free" already implies that space is "kept free", of course by removing stuff.

---

> Do you mean that "disabled" would "disable the feature to skip cleanup"?

Yes. With my first approach `0` would retain the current default behavior and with using the free percentage `100` would retain the current default behavior.

"results_min_free_percentage" seems misleading because this doesn't configure something to ensure a minimum free percentage. It configures not to take care if there's more free disk space than the specified percentage. I don't think "cleanup" is redundant. "results_min_free_percentage" sounds like openQA might cease to take more jobs to ensure the min free disk space and it is not necessarily clear that the thresholds is cleanup related.

---

hm. Looking forward for the plans we had I still think "results_min_free_percentage" might be better but in the current implementation of course your suggestion is better fitting. Ok, we can keep it as you proposed and see if later we want to keep it, introduce another setting or add additional ones

---

I see that you changed it now and have used the default 100. As I wrote above I would have accepted your original proposal. And I find the default value of 100 as tricky. If you want to keep the new names how about treating 0% as a special disabled value?

---

100 as default is not really tricky. I've just figured that this was the reason I intuitively make it work the other way around. We can still treat 0 as special value but I'm not sure whether it makes things better.

---

fine. Anyway, looks good now.